### PR TITLE
chore(typings): remove explicit returns, add type to sync_execute

### PR DESCRIPTION
**Pull request by hazzadous**
We add Callable in a number of places, but it would be better to rely on
the implicit return values for functions.

We also have the clickhouse `execute` method that can return None.
Adding an explicit None in `sync_execute` gives us better visibility
over where we are incorrectly typed.

## Changes

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Comment by guidoiaquinti** [comment]
I've added a `# type: ignore` for all the issues in tests file + fixed a couple of issues in non-test files. 

Now we still have ~100 occurrences that we can either continue to ignore and/or fix properly. Not sure if we want to do this in multiple PRs and risk tons of conflicts over time or not.

**Comment by guidoiaquinti** [comment]
After discussing with the team, we've decided to add `# type: ignore` to all the current occurrences as resolving each of them will involve a non-trivial amount of work.

We are ok to do this for now in order to don't introduce any new regression. We will then communicate with feature owners to fix and remove those ignores.

**Comment by hazzadous** [comment]
👍 from me @guidoiaquinti

**Review by ellie — APPROVED** [review]

File: posthog/client.py [hunk 1]
```diff
@@ -148,7 +148,7 @@ def sync_execute(
     flush=True,
     client_query_id: Optional[str] = None,
     client_query_team_id: Optional[int] = None,
-):
+) -> Optional[Any]:
```
Comment (neilkakkar) [inline posthog/client.py:151]: Is the expectation now to explicitly cast the results we get from sync_execute, wherever sync_execute is called?

Comment (guidoiaquinti) [reply posthog/client.py:151]: Yes, as the `clickhouse-driver` doesn't support types we've added this default.

**Review by neilkakkar — APPROVED** [review]
I'm not sure how this will help mypy catch anything?

Since `sync_execute` can pretty much return arbitrarily, depending on the shape of the query, any typing will be explicit from our side (and I'm pretty sure people will forget to make it an `Optional` type). And at that stage, since we're overriding the type ourselves, mypy assumes we know best.

For sure though, being explicit is better than implicit, atleast forces us to think about this & catch during reviews. So, all for it, but don't know if this is the best way to go about it 🤔 .

**Comment by hazzadous** [comment]
Worth doing something narrower first, closing to remove noise

**Pull request closed without merging**